lib/db.js

```diff
@@ -1,15 +1,32 @@
 var Promise = require('bluebird');
 const levelup = require('level');
-const db = Promise.promisifyAll(levelup('./data'));
+const db = Promise.promisifyAll(levelup('./data', {
+	valueEncoding: 'json'
+}));
 
-exports.scores = {
-	get: function(id) {
-		return db.getAsync(id)
-			.then(JSON.parse)
+module.exports = {
+	get: function(id, rev) {
+		return db.getAsync(id + ':' + rev)
+	},
+	revisions: function(id) {
+		return db.getAsync(id + ':revisions')
+			.then(null, function (err) {
+				if (err.notFound) return 0;
+				return Promise.reject(err);
+			})
 	},
 	save: function(id, code, version) {
-		return Promise.resolve({code: code, version: version})
-			.then(JSON.stringify)
-			.then(db.putAsync.bind(db, id))
+		var rev;
+		return this.revisions(id)
+			.then(function (revs) {
+				rev = ++revs;
+				return Promise.join(
+					db.putAsync(id + ':revisions', revs),
```

Comment:
Maybe do these two writes as one transaction at some point. Not a big deal now. https://github.com/Level/levelup#batch